src/main/java/com/alvarium/exampleapp/App.java

```diff
@@ -1,23 +1,55 @@
 package com.alvarium.exampleapp;
 
+import com.alvarium.DefaultSdk;
+import com.alvarium.Sdk;
 import com.alvarium.SdkInfo;
 import com.alvarium.exampleapp.observers.Channel;
+import com.alvarium.annotators.Annotator;
+import com.alvarium.annotators.AnnotatorException;
+import com.alvarium.annotators.AnnotatorFactory;
+import com.alvarium.contracts.AnnotationType;
+import com.alvarium.hash.HashType;
+import com.alvarium.sign.SignatureInfo;
+import com.alvarium.streams.StreamException;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.config.Configurator;
 
 import config.Reader;
 import config.ReaderException;
 import config.ReaderFactory;
 import config.ReaderType;
 
-public class App{
-  public static void main( String[] args ) throws ReaderException {
-    ReaderFactory factory = new ReaderFactory();
-    Reader reader = factory.getReader(ReaderType.JSON);
+public class App {
+  public static void main(String[] args) throws ReaderException, AnnotatorException,
+      StreamException {
+    ReaderFactory readerFactory = new ReaderFactory();
+    Reader reader = readerFactory.getReader(ReaderType.JSON);
     SdkInfo sdkInfo = reader.read("./src/main/resources/config.json");
 
+    // init sdk dependencies
+    final Annotator[] annotators = new Annotator[sdkInfo.getAnnotators().length];
+    final AnnotatorFactory annotatorFactory = new AnnotatorFactory();
+    for (int i = 0; i < annotators.length; i++) {
+      final AnnotationType kind = sdkInfo.getAnnotators()[i];
+      final HashType hash = sdkInfo.getHash().getType(); 
+      final SignatureInfo signature = sdkInfo.getSignature(); 
+      annotators[i] = annotatorFactory.getAnnotator(kind, hash, signature);
```

Comment:
I'm conflicted on this loop. Two things come to mind, the second of which is more important.

1.) Hash and Signature are always the same value and so could be set outside the loop rather than recreated each time.
2.) In the Go version, the AnnotatorFactory takes the [SDK config](https://github.com/project-alvarium/alvarium-sdk-go/blob/16bd9f306d79529ca2144bf3dcf57c1bc66a9f8c/pkg/factories/factory.go#L55) as a parameter. This was done because I have no way of knowing in the future which config-driven options may be required to create future Annotator types. As the number of types increases, so might the number of parameters on `getAnnotator()`. Indeed we may have parameters which don't apply to all types we need to initialize, and I might need to modify this code every time I add a new Annotator. If you pass in the SdkConfig object and then encapsulate its handling inside the factory, your interface becomes less brittle.

The above is an application of what's called the `Open-Closed Principle`. More info [here](http://joelabrahamsson.com/a-simple-example-of-the-openclosed-principle/). If you're not familiar with the [SOLID](https://www.bmc.com/blogs/solid-design-principles/) principles, take a look.

Reply:
@tsconn23 I've actually mentioned this a while back in this [PR](https://github.com/project-alvarium/alvarium-sdk-java/pull/34#discussion_r702412035) when working on the `AnnotatorFactory`. It was expected to swap out the `HashType hash, SignatureInfo signature` dependencies with `SdkInfo`, but I totally forgot to make an issue for it. I remember this was conflicting with existing PRs so I postponed the implementation for a bit. That was my bad. I've made a [PR](https://github.com/project-alvarium/alvarium-sdk-java/pull/78) on the sdk repo to fix this. Once it's merged, I'll rebase and fix this branch.

as for your first point, the variables should not be needed by now, we'll just pass `SdkInfo`.

Reply:
@tsconn23 fixed in eb0fa310f187caabc389653f20e74c7675d8f1a1